pkg/provider/azure_storageaccount.go

```diff
@@ -355,6 +360,34 @@ func (az *Cloud) EnsureStorageAccount(ctx context.Context, accountOptions *Accou
 			return "", "", fmt.Errorf("failed to create storage account %s, error: %v", accountName, rerr)
 		}
 
+		if pointer.BoolDeref(accountOptions.EnableBlobVersioning, false) || accountOptions.SoftDeleteBlobs > 0 || accountOptions.SoftDeleteContainers > 0 {
+			var blobPolicy, containerPolicy *storage.DeleteRetentionPolicy
+			if accountOptions.SoftDeleteContainers > 0 {
+				containerPolicy = &storage.DeleteRetentionPolicy{
+					Enabled: pointer.Bool(accountOptions.SoftDeleteContainers > 0),
+					Days:    pointer.Int32(accountOptions.SoftDeleteContainers),
+				}
+			}
+			if accountOptions.SoftDeleteBlobs > 0 {
+				blobPolicy = &storage.DeleteRetentionPolicy{
+					Enabled: pointer.Bool(accountOptions.SoftDeleteBlobs > 0),
+					Days:    pointer.Int32(accountOptions.SoftDeleteBlobs),
+				}
+			}
+
+			property := storage.BlobServiceProperties{
+				BlobServicePropertiesProperties: &storage.BlobServicePropertiesProperties{
+					IsVersioningEnabled:            accountOptions.EnableBlobVersioning,
```

Comment:
avoid using `*bool` directly since next time if `accountOptions.EnableBlobVersioning` changes, it may change, define a new bool var here is safer.

Reply:
Do you mean we should copy value from `accountOptions.EnableBlobVersioning` rather then pass the pointer itself directly?

Reply:
like this, it's safer:
```
value := *accountOptions.EnableBlobVersioning
IsVersioningEnabled: &value
```

Reply:
```
var value *bool
if accountOptions.EnableBlobVersioning != nil {
  value = *accountOptions.EnableBlobVersioning
}
...
IsVersioningEnabled: &value
```

Reply:
ok